frontend/src/App.tsx

```diff
@@ -1,15 +1,168 @@
 import { Fragment, useRef, useState } from 'react'
 import { Dialog, Transition } from '@headlessui/react'
 
-import { useGetModelsQuery } from './services/baseService'
+import { useGetModelsQuery, useRegisterModelMutation } from './services/baseService'
 import { CloudIcon, BookOpenIcon } from '@heroicons/react/24/solid'
+import { ModelInfo, ModelType } from './api'
+
+
+function RegisterModelForm({
```

Comment:
This was more or less just extracted from the old inline block. Lets us keep the state internal to the component